src/averageAges.js

```diff
@@ -53,9 +72,35 @@ function calculateWomenAverageAge(people, withChildren) {
  * @param {boolean} onlyWithSon - optional
  *
  * @return {number}
+ *
+ *
+ *
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const women = people.filter(person => person.sex === 'f');
+
+  const ages = women.map(woman => {
+    let womansChildren = [];
+
+    onlyWithSon
+      ? womansChildren = people.filter(person => person.mother === woman.name
+        && person.sex === 'm')
+      : womansChildren = people.filter(person => person.mother === woman.name);
+
+    const childrenYears = womansChildren.map(children => children.born);
+
+    return {
+      womanYear: woman.born,
+      childrenYears,
```

Comment:
It would be more clear if you considered naming object keys differently. I guess womanBornYear and childrenBornYears are suitable here.